core/src/main/java/hudson/cli/BuildCommand.java

```diff
@@ -157,7 +156,7 @@ protected int run() throws Exception {
 
         if (!job.isBuildable()) {
             String msg = Messages.BuildCommand_CLICause_CannotBuildUnknownReasons(job.getFullDisplayName());
-            if (job instanceof AbstractProject<?, ?> && ((AbstractProject<?, ?>)job).isDisabled()) {
+            if (job.isDisabled()) {
```

Comment:
This seems like an (existing) failure of abstraction. There should be a getter for an explanation of why the job is not buildable. Compare `Run.getWhyKeepLog()`.